src/core/reference/include/ngraph/runtime/reference/unique.hpp

```diff
@@ -113,16 +113,24 @@ UniqueElements<Index_t, Count_t> find_unique_elements(const Data_t* data,
         return *(data + lhs.idx) < *(data + rhs.idx);
     };
 
+    int64_t axisVal = 0;
+    if (axis) {
+        axisVal = *axis;
+        if (axisVal < 0) {
+            axisVal += data_shape.size();
+        }
+    }
+
```

Comment:
@tomdol Looks like negative axis should be covered by Unique reference implementation and the reference tests as well.